### PR TITLE
Homework python 29 30

### PR DESCRIPTION
**Pull request by EfimovEO**
Homework python 29 30 done 04042021 1526

### CODE REVIEW

File: python/hw_29-30_task-1.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+Task 1
+Self-study input() function. Write a script which accepts a sequence of comma-separated numbers from user 
+and generate a list and a tuple with those numbers and prints these objects as-is (just print(list) without any formatting).
+'''
+
+x = input("Please, input comma-separated numbers:")
+ls = x.split(',')
+print("This is list: {}".format(ls))
+tp=tuple(ls)
+print("This is tuple: {}".format(tp))
```
[the remark this answers is not in the document]
Comment (EfimovEO) [reply python/hw_29-30_task-1.py]: of course. done

File: python/hw_29-30_task-2.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    try:
+        a=int(i)
+    except ValueError:
+        print("{} - this is not number".format(i))
+    else:
+        if a==254:
+            print("254 was found")
+            break
+        else:
+            print(a)
```
[the remark this answers is not in the document]
Comment (EfimovEO) [reply python/hw_29-30_task-2.py]: done

File: python/hw_29-30_task-3.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+args_list.pop(0)
+x = input("Please, input some word or phrase: ")
+y=x.split(' ')
+while True:
+    if check(args_list, y):
+        print("You have provided one of argument phrases.")
+        break
+    else:
+        x = input("Please, input some word or phrase one more time: ")
+        y=x.split(' ')
```
[the remark this answers is not in the document]
Comment (EfimovEO) [reply python/hw_29-30_task-3.py]: done

File: python/hw_29-30_task-4.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+pid = os.getpid()
+usename=pwd.getpwuid(os.getuid())[0]
+OS_NAME=os.uname()[0]
+OS_RELEASE=os.uname()[2]
+tx='''
+This script has the following PID: {0}.
+It was ran by {1}
+to work happily on {2}-{3}.
+'''.format(pid, usename, OS_NAME, OS_RELEASE)
+print(tx)
```
[the remark this answers is not in the document]
Comment (EfimovEO) [reply python/hw_29-30_task-4.py:25]: 1.done
2.add one more way

File: python/hw_29-30_task-5.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+ls=list(myset)
+ls.sort(reverse=True)
+
+if len(ls)<3:
+    top=len(ls)
+else:
+    top=3
+
+print("List of top max integers: {}".format(ls[0:top]))
+print("You've passed some extra elements that I can't parse: {}".format(error_ls))
```
[the remark this answers is not in the document]
Comment (EfimovEO) [reply python/hw_29-30_task-5.py]: done

File: python/hw_29-30_task-6.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+Task 6
+Create a function that will take a string as an argument and return a dictionary
+where keys are symbols from the string and values are the count of inclusion of that symbol.
+'''
+
+
+x = input("Please, input a string:")
+ls = list(x)
+dictionary={i:ls.count(i) for i in set(ls)}
+print(dictionary)
```
[the remark this answers is not in the document]
Comment (EfimovEO) [reply python/hw_29-30_task-6.py]: done

File: python/hw_29-30_task-7.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+y = 0
+
+# эти переменные задают движение по матрице.
+dx = 1
+dy = 0
+
+#заполнение матрицы
+for i in range(N * N):
+	a[y][x] = i + 1
+    ts = x + dx if dx else y + dy
```
[the remark this answers is not in the document]
Comment (EfimovEO) [reply python/hw_29-30_task-7.py]: 1. was the problem with tabs and spaces
2. done

File: python/hw_29-30_task-9.py [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+4. container information, like docker inspect 
+ 
+Connection function must accept connection string for example 'http://192.168.56.101:2376' and connect to it or use string from environment
+if no connection string is given.    
+In order to connect to docker, you can use either Unix socket or reconfigure daemon to use a network socket 
+(https://docs.docker.com/engine/reference/commandline/dockerd/#daemon-socket-option)
+'''
+import sys
+from pprint import pprint
+import docker
```
[the remark this answers is not in the document]
Comment (EfimovEO) [reply python/hw_29-30_task-9.py:18]: Do you mean Go and HTTP?

From: https://docs.docker.com/engine/api/sdk/

Docker provides an API for interacting with the Docker daemon (called the Docker Engine API), as well as SDKs for Go and Python. The SDKs allow you to build and scale Docker apps and solutions quickly and easily. If Go or Python don’t work for you, you can use the Docker Engine API directly.

The Docker Engine API is a RESTful API accessed by an HTTP client such as wget or curl, or the HTTP library which is part of most modern programming languages.

File: python/hw_29-30_task-9.py [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+for container in client.containers.list(all=True):
+    print('\n\n\nInsperct of {}.\n\n\n'.format(container.name))
+    pprint(client_api.inspect_container(container.id))
+    input("Press ENTER to continue.")
+
+
+
+
+client.close()
+client_api.close()
```
[the remark this answers is not in the document]
Comment (EfimovEO) [reply python/hw_29-30_task-9.py]: done

File: python/hw_29-30_task-9.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (Cakemanator) [reply python/hw_29-30_task-9.py:18]: https://docs.docker.com/engine/api/sdk/examples/
`curl --unix-socket /var/run/docker.sock http://localhost/v1.41/containers/json`
You can use the `requests` module to perform such types of requests and get the information you need.

[the remark this answers is not in the document]
Comment (EfimovEO) [reply python/hw_29-30_task-9.py:18]: was added on more  function which interacting with the Docker daemon by python requests

**Pull request merged**